plugin/type.go

```diff
@@ -27,22 +27,22 @@ type ComposePlugin interface {
 	Name() string
 
 	// execute some tasks before the Image is pulled
-	LaunchTaskPreImagePull(ctx *context.Context, composeFiles *[]string, executorId string, taskInfo *mesos.TaskInfo) error
+	LaunchTaskPreImagePull(ctx context.Context, composeFiles *[]string, executorId string, taskInfo *mesos.TaskInfo) error
 
 	// execute some tasks after the Image is pulled
-	LaunchTaskPostImagePull(ctx *context.Context, composeFiles *[]string, executorId string, taskInfo *mesos.TaskInfo) error
+	LaunchTaskPostImagePull(ctx context.Context, composeFiles *[]string, executorId string, taskInfo *mesos.TaskInfo) error
 
 	// execute the tasks after the pod is launched
-	PostLaunchTask(ctx *context.Context, composeFiles []string, taskInfo *mesos.TaskInfo) (string, error)
+	PostLaunchTask(ctx context.Context, composeFiles []string, taskInfo *mesos.TaskInfo) (string, error)
 
 	// execute the task before we send a Kill to Mesos
-	PreKillTask(taskInfo *mesos.TaskInfo) error
+	PreKillTask(ctx context.Context, taskInfo *mesos.TaskInfo) error
 
 	// execute the task after we send a Kill to Mesos
-	PostKillTask(taskInfo *mesos.TaskInfo) error
+	PostKillTask(ctx context.Context, taskInfo *mesos.TaskInfo) error
 
 	// execute the task to shutdown the pod
-	Shutdown(executor.ExecutorDriver) error
+	Shutdown(taskInfo *mesos.TaskInfo, ed executor.ExecutorDriver) error
```

Comment:
Is it still required to add `taskInfo ` in this interface?

Reply:
For our side, it is. Because we need taskInfo to re-construct the root context.

@vipragupta But I found, the Shutdown implementation in all each plugins don't have any functionality, just print an line. I think maybe we can skip have a wrapper of this implementation?